flink/v1.19/flink/src/main/java/org/apache/iceberg/flink/sink/shuffle/MapDataStatistics.java

```diff
@@ -22,49 +22,69 @@
 import org.apache.flink.annotation.Internal;
 import org.apache.iceberg.SortKey;
 import org.apache.iceberg.relocated.com.google.common.base.MoreObjects;
+import org.apache.iceberg.relocated.com.google.common.base.Objects;
 import org.apache.iceberg.relocated.com.google.common.collect.Maps;
 
 /** MapDataStatistics uses map to count key frequency */
 @Internal
-class MapDataStatistics implements DataStatistics<MapDataStatistics, Map<SortKey, Long>> {
-  private final Map<SortKey, Long> statistics;
+public class MapDataStatistics implements DataStatistics {
```

Comment:
nit: Why is this public?

Reply:
you are right. this was public with the generics approach. but now the implementation classes can be package private